core/src/main/java/io/grpc/transport/AbstractStream.java

```diff
@@ -157,22 +165,23 @@ public int getOnReadyThreshold() {
    */
   public void setOnReadyThreshold(int onReadyThreshold) {
     checkArgument(onReadyThreshold > 0, "onReadyThreshold must be > 0");
-    boolean doNotify;
+    boolean doNotify = false;
     synchronized (onReadyLock) {
-      if (this.onReadyThreshold <= numSentBytesQueued && onReadyThreshold > numSentBytesQueued) {
-        shouldNotifyOnReady = true;
-      }
+      doNotify = wentBelowThreshold(
+          numSentBytesQueued,
+          this.onReadyThreshold, // old threshold
+          numSentBytesQueued,
+          onReadyThreshold); // new threshold.
       this.onReadyThreshold = onReadyThreshold;
-      doNotify = needToNotifyOnReady();
```

Comment:
Hmm, I thought I pressed reply, but I guess not.   My original response:

I was hoping onReady() would only be called from one place, notify if ready.  If the stream has not yet been allocated, it would be wrong to called onReady() directly from here.  (or if the listener has not yet been allocated).  I believe this method was only intended to be called from tests, but I wasn't positive about that.  Calling through notifyIfReady allows this to be called from prod code as well.